test/test_write.ml

```diff
@@ -0,0 +1,120 @@
+open Icalendar
+
+let empty = Params.empty
+
+let to_ptime date time =
+  match Ptime.of_date_time (date, (time, 0)) with
+  | None -> Alcotest.fail "invalid date time"
+  | Some p -> p
+
+let singleton k v = Params.add k v empty
+
+let test_serialize_calendar () =
+  let expected =
+    {|BEGIN:VCALENDAR
+VERSION:2.0
+CALSCALE:GREGORIAN
+BEGIN:VEVENT
+CREATED:20191109T223039Z
+LAST-MODIFIED:20191109T223039Z
+DTSTAMP:20191109T223039Z
+UID:ulre32v9-29gj-5k2h-me3x-eu4d4l7d316y
+SUMMARY:djlkfjklsfs
+TRANSP:OPAQUE
+CLASS:PUBLIC
+DTSTART;VALUE=DATE:20191104
+DTEND;VALUE=DATE:20191105
+EXDATE:20240301T000000Z,20240303T000000Z,20240305T000000Z
+END:VEVENT
+PRODID:-//Inf-IT//CalDavZAP 0.13.1//EN
+END:VCALENDAR
+|}
+
+  and input =
```

Comment:
Using `and` in `let f1 = ... and f2 = ... and f3 = ...` actually allows `f1`, `f2` and `f3` to be mutually recursive. IMO it makes the code harder to read because you are expecting to see some complex mutually recursive stuff going one but there is none. Using `let f1 = ... in let f2 = ... in ...` takes a little bit more character but it makes also the intend much more clearer for future reader.

Reply:
Thank you. I don't mind typing more :)